src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,143 @@
+    );
+    // activate AutoRemoteFollow
+    window.open(`/interact/${tootId}`, "_blank");
+}
+
+/**
+ * Wait for element to appear.
+ * 
+ * @param {string} selector
+ * @param {boolean} multiple
+ * @param {number} timeoutDuration
+ * @see {@link https://github.com/storybookjs/test-runner/blob/6d41927154e8dd1e4c9e7493122e24e2739a7a0f/src/setup-page.ts#L134}
+ *  from which this was adapted
+ * @returns {Promise}
+ */
+function waitForElement(selector, multiple = false, timeoutDuration) {
+    return new Promise((resolve, reject) => {
+        const getElement = () => (
+            multiple
+                ? document.querySelectorAll(selector)
+                : document.querySelector(selector)
+        );
+        const isElementFound = (el) => (!multiple && el) || (multiple && el.length > 0);
+
+        const timeout = window.setTimeout(() => {
+            reject(new Error("waitForElement timed out"));
+        }, timeoutDuration);
+
+        const element = getElement();
+        if(isElementFound(element)){
+            window.clearTimeout(timeout);
+            return resolve(element);
+        }
+
+        const observer = new MutationObserver(() => {
+            const element = getElement();
+            if(isElementFound(element)){
+                window.clearTimeout(timeout);
+                resolve(element);
+                observer.disconnect();
+            }
+        });
+
+        observer.observe(document.body, {
+            childList: true,
+            subtree: true
+        });
+
+        return null;
+    }); 
+}
+
+/**
+ * Inject replacement onClick handler for Follow button.
+ * 
+ * @returns {void}
+ */
+async function injectFollowButton() {
+    try {
+        const followButton = await waitForElement(".account__header__tabs__buttons button:first-of-type", false, TIMEOUT_DURATION);
+        followButton.addEventListener("click", onClickFollow);
+    } catch (error) {
+        // Follow button failed to appear
+    }
+}
+
+/**
+ * Inject replacement onClick handler for Interaction buttons.
+ * 
+ * @returns {void}
+ */
+async function injectInteractionButtons() {
+    const INJECTED_REPLY_CLASS = "mastodon-simplified-federation-injected-interaction";
+    try {
+        const replyButtons = await waitForElement(
+            ".item-list[role='feed'] article[data-id] .status__action-bar button," +
+            ".detailed-status__wrapper .detailed-status__action-bar button",
+            true,
+            TIMEOUT_DURATION,
+        );
+        replyButtons.forEach((button) => {
+            if(!button.classList.contains(INJECTED_REPLY_CLASS)){
+                button.addEventListener("click", onClickInteract);
+                button.classList.add(INJECTED_REPLY_CLASS);
+            }
+        });
+    } catch (error) {
+        // Interaction buttons failed to appear
+        console.log(error);
+    }
+}
+
+/**
+ * Initialise injection for Mastodon Follow button.
```

Comment:
```suggestion
 * Initialise injection for all remote Mastodon buttons.
```

It's not just about the follow button now.

Reply:
Very true, thanks for catching that.